test/parallel/test-worker-dispose.mjs

```diff
@@ -0,0 +1,10 @@
+import * as common from '../common/index.mjs';
+import { Worker } from 'node:worker_threads';
+
+let w;
+{
+  // Verifies that the worker is async disposable
+  await using worker = new Worker('for(;;) {}', { eval: true });
+  w = worker;
+  w.on('exit', common.mustCall());
+}
```

Comment:
Nit: was `w` meant to be checked after the scope? If not, we don't really need it.
```suggestion
{
  // Verifies that the worker is async disposable
  await using worker = new Worker('for(;;) {}', { eval: true });
  worker.on('exit', common.mustCall());
}
```